src/main/python/mlsearch/api_requester.py

```diff
@@ -248,6 +250,13 @@ def _fetch_youtube(self, y_next_page_token=None) -> [Protocol]:
         sampled_dev_key = None
         if len(self._config.YOUTUBE_DEVELOPER_KEY) > 0:
             sampled_dev_key = random.choice(self._config.YOUTUBE_DEVELOPER_KEY)
+
+        if not sampled_dev_key:
+            auth_error = ErrorType(
+                reason="Empty YouTube Developer Key.", status="400"
+            )
+            raise HttpError(auth_error, str.encode("YouTube Developer Key Required."))
+
```

Comment:
Sorry, for being too detail. 
Actually, I was thinking simpler one like

```suggestion
         if not len(self._config.YOUTUBE_DEVELOPER_KEY) > 0:
            auth_error = ErrorType(
                reason="Empty YouTube Developer Key.", status="400"
            )
            raise HttpError(auth_error, str.encode("YouTube Developer Key Required."))

         sampled_dev_key = random.choice(self._config.YOUTUBE_DEVELOPER_KEY)
```

How do you think? It'll remove one `if` 😄

Reply:
Thanks... I fixed it.